examples/advanced/llm_hf/code/hf_sft_peft_fl.py

```diff
@@ -0,0 +1,227 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import copy
+import os
+
+import datasets
+import torch
+from peft import LoraConfig, get_peft_model, get_peft_model_state_dict, set_peft_model_state_dict, utils
+from transformers import AutoModelForCausalLM, AutoTokenizer, TrainingArguments, trainer_utils
+from trl import SFTTrainer
+
+import nvflare.client as flare
+
+use_flash_attention = True
+
+
+def format_instruction(example):
+    output_texts = []
+    for i in range(len(example["input"])):
+        text = f"### Instruction: Generate Output according to the information and question given by Input. ### Input:{example['input'][i]} ### Response: {example['output'][i]}"
+        output_texts.append(text)
+    return output_texts
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--model_path",
+        type=str,
+        default="./model/Llama-2-7b-hf",
+    )
+    parser.add_argument(
+        "--data_path_train",
+        type=str,
+        default="./dataset/dolly/training.jsonl",
+    )
+    parser.add_argument(
+        "--data_path_valid",
+        type=str,
+        default="./dataset/dolly/validation.jsonl",
+    )
+    parser.add_argument(
+        "--output_path",
+        type=str,
+        default="llama2-7b-dolly-sft",
+    )
+    parser.add_argument("--mode", type=int, default=0)
```

Comment:
Could use a description for this option. Why not use --peft as arg and and set it as a boolean flag directly, i.e. `parser.add_argument('--peft', action='store_true')?